Marlin/src/lcd/extui/lib/dgus_creality/creality_touch/PageHandlers.cpp

```diff
@@ -0,0 +1,350 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2020 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+#include "../../../../../inc/MarlinConfigPre.h"
+
+#if ENABLED(DGUS_LCD_UI_CREALITY_TOUCH)
+
+#include "../DGUSDisplayDef.h"
+#include "../DGUSDisplay.h"
+#include "../DGUSScreenHandler.h"
+
+#include "../../../../../module/temperature.h"
+#include "../../../../../module/motion.h"
+#include "../../../../../module/planner.h"
+#include "../../../../../feature/pause.h"
+#include "../../../../../feature/runout.h"
+#include "../../../../../module/settings.h"
+
+#include "../../../../marlinui.h"
+#include "../../../ui_api.h"
+
+#include "PageHandlers.h"
+
+// Definitions of page handlers
+
+void MainMenuHandler(DGUS_VP_Variable &var, unsigned short buttonValue) {
+  // Nothing
+}
+
+void ControlMenuHandler(DGUS_VP_Variable &var, unsigned short buttonValue) {
+  switch (var.VP) {
+    case VP_BUTTON_PREPAREENTERKEY:
+      switch (buttonValue) {
+        case 5:       // About
+          // Automatically handled
+          break;
+
+        case 7:       // Reset to factory settings
+          settings.reset();
+          settings.save();
+          break;
+
+        case 9:       // Back button
+          // TODO: should navigate automatically
+          break;
+      }
+      break;
+
+    case VP_BUTTON_ADJUSTENTERKEY:
+      ScreenHandler.HandleLEDToggle();
+      break;
+  }
+}
+
+void LevelingModeHandler(DGUS_VP_Variable &var, unsigned short buttonValue) {
+  switch (var.VP) {
+    case VP_BUTTON_BEDLEVELKEY:
+      switch (buttonValue) {
+        case 1:
+          if (ExtUI::isAxisPositionKnown(ExtUI::axis_t::X) && ExtUI::isAxisPositionKnown(ExtUI::axis_t::Y))
+            ExtUI::injectCommands_P("G28 Z");
+          else
+            ExtUI::injectCommands_P("G28");
+          break;
+
+        case 2:
+          // Increase Z-offset
+          ExtUI::smartAdjustAxis_steps(5, ExtUI::axis_t::Z, true);
```

Comment:
Needs to be adjusted via `ExtUI::mmToWholeSteps`: https://github.com/CR6Community/Marlin/commit/583a7f25186ded1c05cbd25b3d0ba1480a70f2de